source/falaise/snemo/services/dead_cells.h

```diff
@@ -0,0 +1,128 @@
+//! \file falaise/snemo/services/dead_cells.h
+//! \brief Tracker Dead Cells service in SuperNEMO
+//
+// more doxygen to come...
+//
+// status  description
+// ---------------------------------------------------------------------------
+// 0       good working cell            isAGoodCell==true
+// 1       dead cells with dead anode   isADeadCell==true and isABadCell==true
+// 2       cathode-to-ground short      isABadCell==true
+// 3       cathode-to-cathode short     isABadCell==true
+// 4       other                        isABadCell==true
+
+
+#ifndef SNEMO_DEAD_CELLS_SVC_HH
+#define SNEMO_DEAD_CELLS_SVC_HH
+
+#include <iostream>
+#include <fstream>
+
+#include "bayeux/datatools/base_service.h"
+#include "falaise/snemo/services/service_traits.h"
+#include "falaise/bounded_int.h"
+
+namespace snemo {
+
+  // data member for the possible statuses of a cell  
+  enum status {
+    good = 0,            // good working cell
+    dead = 1,            // dead cells with dead anode
+    cathode_ground = 2,  // cathode-to-ground short
+    cathode_cathode = 3, // cathode-to-cathode short
+    other = 4            // other
+  };
+
+  // data member defining a cell structure
+  class cell_id {
+  public:
+    using side_id = falaise::bounded_int<0,1>;
+    using layer_id = falaise::bounded_int<0,8>;
+    using column_id = falaise::bounded_int<0,112>;
+    using status_id = falaise::bounded_int<0,4>;
+
+    // constructors
+    cell_id(side_id s, layer_id l, column_id c){  // use this to search for the status of a cell
+      SetCell(s,l,c);
+    }
+
+    cell_id(side_id s, layer_id l, column_id c, status_id st){ // use this to create a cell knowing everything
+      SetCell(s,l,c,st);
+    }
+    
+    // functions
+    void SetCell(side_id s, layer_id l, column_id c){  // use this to search for the status of a cell
+      side_ = s;
+      layer_ = l;
+      column_ = c;
+    }
+
+    void SetCell(side_id s, layer_id l, column_id c, status_id st){  // use this to create a cell knowing everything
+      side_ = s;
+      layer_ = l;
+      column_ = c;
+      status_ = st;
+    }
+
+    // accessors
+    int GetSide(){
+      return side_;
+    }
+
+    int GetLayer(){
+      return layer_;
+    }
+
+    int GetColumn(){
+      return column_;
+    }
+
+    int GetStatus(){
+      return status_;
+    }
+
+  private:
+    side_id side_;
+    layer_id layer_;
+    column_id column_;
+    status_id status_;
+  };
+  
+  class dead_cells_svc : public datatools::base_service {
+  public:
+
+    // Vector with list of Bad cells to be filled by the function LoadCells()
+    std::vector<cell_id> cells;    
```

Comment:
Should be `private`?

Reply:
I think could be useful to access this vector from outside without having to re-create one (e.g. want to know how many bad cells are there with .size() or loop over them).

Reply:
In that case, it's probably better to add those interfaces to the class, e.g.

```c++
size_t size() const { return cells.size();}

std::vector<cell_id>::const_iterator begin() const { return cells.begin();}

std::vector<cell_id>::const_iterator end() const { return cells.end();}
```

or similar. This should also prevent accidental rewrites without going through the appropriate interface.

Reply:
Oh, and it's fine to add some `using`/`typedefs` if you want to simplify the return type!